client/client.go

```diff
@@ -145,54 +52,52 @@ func (c *Client) close() error {
 	return nil
 }
 
-func (c *Client) ExecuteProtocol(params ProtocolParams) {
-	schemaType := pb.SchemaType_name[int32(c.schema)]
-	schemaVariant := pb.SchemaVariant_name[int32(c.variant)]
-	variant := common.ToProtocolType(c.variant)
+func NewGenericClient(endpoint string) (*genericClient, error) {
```

Comment:
Indeed. I removed export of generic client initialization function (now it's named `newGenericClient` as you suggested).

Personally, I prefer to keep the `struct`s on top before initializer functions. But indeed I wasn't being 100% consistent. I used your suggested template and moved auxiliary function `validateSchema` in `pedersen.go` to the bottom. I also modified `client.go` to follow this template.